src/util-log-kafka.c

```diff
@@ -0,0 +1,203 @@
+    }
+
+    if (kafka_ctx->rk) {
+    	/* Destroy the handle */
+    	rd_kafka_destroy(kafka_ctx->rk);
+    }
+    return;
+}
+
+/**
+ * \brief LogFileWriteKafka() writes log data to kafka output.
+ * \param lf_ctx Log file context allocated by caller
+ * \param string buffer with data to write
+ * \param string_len data length
+ * \retval 0 on sucess;
+ * \retval -1 on failure;
+ */
+int LogFileWriteKafka(void *lf_ctx, const char *string, size_t string_len)
+{    
+    LogFileCtx *log_ctx = lf_ctx;    
+    SCLogKafkaContext *kafka_ctx = log_ctx->kafka;
+    int partition = kafka_ctx->partition % (log_ctx->kafka_setup.partitions);
+    
+    if (rd_kafka_produce(kafka_ctx->rkt, partition,
+			RD_KAFKA_MSG_F_COPY,
+			/* Payload and length */
+			(void *)string, string_len,
+			/* Optional key and its length */
+			NULL, 0,
+			/* Message opaque, provided in
+			 * delivery report callback as
+			 * msg_opaque. */
+			NULL) == -1) 
+	{
+		SCLogError(SC_ERR_KAFKA,
+				"%% Failed to produce to topic %s "
+				"partition %i: %s\n",
+				log_ctx->kafka_setup.topic_name, partition,
+				rd_kafka_err2str(
+					rd_kafka_errno2err(errno)));
+		/* Poll to handle delivery reports */
+		rd_kafka_poll(kafka_ctx->rk, 0);
+	}
+    kafka_ctx->partition++;
+    
+    return -1;
+}
+
+/**
+ * \brief Message delivery report callback.
+ * Called once for each message.
+ */
+static void msg_delivered (rd_kafka_t *rk,
+			   void *payload, size_t len,
+			   int error_code,
+			   void *opaque, void *msg_opaque) 
+{
+    rk = rk;
+    payload = payload;
+    len = len;
+    opaque = opaque;
+    msg_opaque = msg_opaque;
+    if (error_code)
+        SCLogError(SC_ERR_KAFKA, "%% Message delivery failed: %s\n",
+            rd_kafka_err2str(error_code));
+}
+
+/** \brief configure and initializes kafka output logging
+ *  \param kafka_node ConfNode structure for the output section in question
+ *  \param lf_ctx Log file context allocated by caller
+ *  \retval 0 on success
+ */
+int SCConfLogOpenKafka(ConfNode *kafka_node, void *lf_ctx)
+{
+    LogFileCtx *log_ctx = lf_ctx;
+    const char *partitions = NULL;
+    SCLogKafkaContext *kafka_ctx = NULL;
+
+    if (NULL == kafka_node) {
+        return -1;
+    }
+    
+    log_ctx->kafka_setup.brokers = ConfNodeLookupChildValue(kafka_node, "brokers");
+    log_ctx->kafka_setup.topic_name = ConfNodeLookupChildValue(kafka_node, "topic");
+    partitions =  ConfNodeLookupChildValue(kafka_node, "partitions");
+    log_ctx->kafka_setup.partitions = atoi(partitions);
+
+    /*create kafka ctx*/        
+    rd_kafka_conf_t *conf;
+    rd_kafka_topic_conf_t *topic_conf;        
+    char tmp[16];        
+    char errstr[512];
+    kafka_ctx = (SCLogKafkaContext*) SCCalloc(1, sizeof(SCLogKafkaContext));
+    if (kafka_ctx == NULL) {
+        SCLogError(SC_ERR_MEM_ALLOC, "Unable to allocate kafka context");
+        exit(EXIT_FAILURE);
+    }
+    
+    conf = rd_kafka_conf_new();

```

Comment:
Make sure you free `conf` with `rd_kafka_topic_conf_destroy` on error paths/early exits.